### DISCUSSION TITLE
umi4 按照默认的分包策略打包的js会放到head里面异步加载，是否可以去掉，变成按需加载

### DISCUSSION THREAD

**Discussion by zhongbo320** [umijs/umi#12618] [Q&A]
umi3 升级 umi4过程中
有这样一个文件 p__QualityCheck__QualityCheckOrder__xxxxx.js 会插入到head中，<script src="/p__QualityCheck__QualityCheckOrder__xxxxx.async.js" async="" data-webpack="as-repair-fe:9692"></script>

实际这个js模块中有代码需等待初始化完成才能执行，如获取远端多语言，此时就会出错，请问是否可以将/p__QualityCheck__QualityCheckOrder__xxxxx.async.js 变为按需加载

**Comment by zhongbo320104** [comment]
已处理，通过写一个插件解决
